Final copy update for DSA notifications

nick-funk opened the pull request:
## What does this PR do?



## These changes will impact:

- [X] commenters
- [X] moderators
- [ ] admins
- [ ] developers

## What changes to the GraphQL/Database Schema does this PR introduce?

None

## Does this PR introduce any new environment variables or feature flags?

No

## If any indexes were added, were they added to `INDEXES.md`?

No new indices

## How do I test this PR?

- enable DSA reporting flow via Admin > Config > General > DSA
- create a comment with a commenter
- report it with an illegal report (DSA report) as the same commenter
- use an admin/mod to process the DSA report as illegal
- check the commenter's stream side notifications
    - should have a notification about the comment being rejected
    - should have a notification about their DSA report being decided/reviewed
- create another comment as commenter
- have an admin/mod reject it normally via mod queue while selecting a reason
- check the notifications for the commenter
    - should have a notification about the comment being rejected

## Where any tests migrated to React Testing Library?

No

## How do we deploy this PR?

Merge into DSA epic branch and release with rest of DSA features.

nick-funk committed:
add method of redress to DSA tenant settings

nick-funk committed:
preliminarily show different icons for notification types

nick-funk committed:
update copy for dsa report decision notifications

nick-funk committed:
preliminarily add copy for illegal rejection notifications

nick-funk committed:
use new copy for general comment rejection notifications

nick-funk committed:
plumg rejectionReason and decisionDetails into notifications

nick-funk committed:
handle rejected comment notification formatting stream side

nick-funk committed:
format dsa report decision notifications stream side

nick-funk committed:
remove title and body from notifications

we are no longer using these as we infer the notification
content stream side based on the state and type of the
notification fields.

kabeaty reviewed (commented):
Took a look at the in progress updates to notifications here, and it's looking really nice so far!

nick-funk committed:
disable approve and feature comment rejection notifications

we aren't doing this for now, we'll get to them later with
the general notifications, we only want rejection and DSA
report decision notifications for now.

nick-funk committed:
make linter happy with disabled notifications
